File: Advanced/snakesandladders-cordapp/contracts/src/test/java/net/corda/samples/snl/contracts/ContractTests.java
package net.corda.sample.snl.contracts;

import net.corda.core.identity.CordaX500Name;
import net.corda.sample.snl.contracts.BoardConfigContract;
import net.corda.sample.snl.states.BoardConfig;
import net.corda.testing.core.TestIdentity;
import net.corda.testing.node.MockServices;
import org.junit.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;

import static net.corda.testing.node.NodeTestUtils.transaction;

public class ContractTests {
    private final TestIdentity p1 = new TestIdentity(new CordaX500Name("PL1", "", "IN"));
    private final TestIdentity p2 = new TestIdentity(new CordaX500Name("PL2", "", "IN"));
    private MockServices ledgerServices = new MockServices(new TestIdentity(new CordaX500Name("TestId", "", "IN")));

    private BoardConfig boardConfig_f = new BoardConfig(null, null, Arrays.asList(p1.getParty(), p2.getParty()));

    private BoardConfig boardConfig_s = new BoardConfig(new LinkedHashMap(Collections.singletonMap(1, 5)),
            new LinkedHashMap(Collections.singletonMap(1, 5)), Arrays.asList(p1.getParty(), p2.getParty()));

    @Test
    public void testVerifyCreateBoardZeroInputs() {
        transaction(ledgerServices, tx -> {
            // Has an input, will fail.
            tx.input(BoardConfigContract.ID, boardConfig_s);
            tx.output(BoardConfigContract.ID, boardConfig_s);
            tx.command(Arrays.asList(p1.getPublicKey(), p2.getPublicKey()), new BoardConfigContract.Commands.Create());
            tx.fails();
            return null;
        });

        transaction(ledgerServices, tx -> {
            // Has no input, will verify.
            tx.output(BoardConfigContract.ID, boardConfig_s);
            tx.command(Arrays.asList(p1.getPublicKey(), p2.getPublicKey()), new BoardConfigContract.Commands.Create());
            tx.verifies();
            return null;
        });
    }

    @Test
    public void testVerifyCreateBoardNonEmptySnakeAndLadderPositions() {
        transaction(ledgerServices, tx -> {
            // Has empty/ null snake and ladder positions, should fail
            tx.output(BoardConfigContract.ID, boardConfig_f);
            tx.command(Arrays.asList(p1.getPublicKey(), p2.getPublicKey()), new BoardConfigContract.Commands.Create());
            tx.fails();
            return null;
        });

        transaction(ledgerServices, tx -> {
            // Has no input, will verify.
            tx.output(BoardConfigContract.ID, boardConfig_s);
            tx.command(Arrays.asList(p1.getPublicKey(), p2.getPublicKey()), new BoardConfigContract.Commands.Create());
            tx.verifies();
            return null;
        });
    }

}

File: kotlin-samples/Advanced/snakesandladders-cordapp/contracts/src/test/kotlin/net/corda/samples/snl/contracts/ContractTests.kt
package net.corda.samples.snl.contracts

import net.corda.core.identity.CordaX500Name
import net.corda.samples.snl.states.BoardConfig
import net.corda.testing.core.TestIdentity
import net.corda.testing.node.MockServices
import net.corda.testing.node.ledger
import org.junit.Test
import java.util.*
import kotlin.collections.LinkedHashMap

class ContractTests {
    private val p1 = TestIdentity(CordaX500Name("PL1", "", "IN"))
    private val p2 = TestIdentity(CordaX500Name("PL2", "", "IN"))
    private val ledgerServices = MockServices(TestIdentity(CordaX500Name("TestId", "", "IN")))

    private val boardConfig_f = BoardConfig(null, null, Arrays.asList(p1.party, p2.party))
    private val boardConfig_s = BoardConfig(LinkedHashMap(Collections.singletonMap(1, 5)),
            LinkedHashMap(Collections.singletonMap(1, 5)), Arrays.asList(p1.party, p2.party))

    @Test
    fun testVerifyCreateBoardZeroInputs() {
        ledgerServices.ledger {
            // Should fail bid price is equal to previous highest bid
            transaction {
                // Has an input, will fail.
                input(BoardConfigContract.ID, boardConfig_s)
                output(BoardConfigContract.ID, boardConfig_s)
                command(Arrays.asList(p1.publicKey, p2.publicKey), BoardConfigContract.Commands.Create())
                fails()
            }
            //pass
            transaction {
                // Has no input, will verify.
                output(BoardConfigContract.ID, boardConfig_s)
                command(Arrays.asList(p1.publicKey, p2.publicKey), BoardConfigContract.Commands.Create())
                verifies()
                verifies()
            }
        }
    }

    @Test
    fun testVerifyCreateBoardNonEmptySnakeAndLadderPositions() {
        ledgerServices.ledger {
            // Should fail bid price is equal to previous highest bid
            transaction {
                // Has empty/ null snake and ladder positions, should fail
                output(BoardConfigContract.ID, boardConfig_f)
                command(Arrays.asList(p1.publicKey, p2.publicKey), BoardConfigContract.Commands.Create())
                fails()
                fails()
            }
            //pass
            transaction {
                // Has no input, will verify.
                output(BoardConfigContract.ID, boardConfig_s)
                command(Arrays.asList(p1.publicKey, p2.publicKey), BoardConfigContract.Commands.Create())
                verifies()
                verifies()
            }
        }
    }
}